drivers/storage/portworx/deployment_test.go

```diff
@@ -1757,6 +1757,117 @@ func TestPodSpecWithCloudStorageSpecOnEKS(t *testing.T) {
 		"-secret_type", "k8s",
 	}
 	actual, _ := driver.GetStoragePodSpec(cluster, fakeK8sNodes.Items[0].Name)
+
+	assert.ElementsMatch(t, expectedArgs, actual.Containers[0].Args)
+
+	// Reset preflight for other tests
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	err = preflight.InitPreflightChecker(k8sClient)
+	require.NoError(t, err)
+}
+
+func TestPodSpecWithCloudStorageSpecOnGCE(t *testing.T) {
+	fakeK8sNodes := &v1.NodeList{Items: []v1.Node{
+		{ObjectMeta: metav1.ObjectMeta{Name: "node1"}, Spec: v1.NodeSpec{ProviderID: "gce://node-id-1"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "node2"}, Spec: v1.NodeSpec{ProviderID: "gce://node-id-2"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "node3"}, Spec: v1.NodeSpec{ProviderID: "gce://node-id-3"}},
+	}}
+
+	versionClient := fakek8sclient.NewSimpleClientset(fakeK8sNodes)
+	versionClient.Discovery().(*fakediscovery.FakeDiscovery).FakedServerVersion = &version.Info{
+		GitVersion: "v1.26.5-gke.1200",
+	}
+	coreops.SetInstance(coreops.New(versionClient))
+	k8sClient := testutil.FakeK8sClient(fakeK8sNodes)
+	err := preflight.InitPreflightChecker(k8sClient)
+	require.NoError(t, err)
+
+	require.True(t, preflight.IsGKE())
+	c := preflight.Instance()
+	require.Equal(t, cloudops.GCE, c.ProviderName())
+
+	driver := portworx{}
+	err = driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(100))
+	require.NoError(t, err)
+
+	cluster := &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1.StorageClusterSpec{
+			Image:        "portworx/oci-monitor:3.0.0",
+			CloudStorage: &corev1.CloudStorageSpec{},
+		},
+	}
+
+	err = driver.SetDefaultsOnStorageCluster(cluster)
+	require.NoError(t, err)
+
+	expectedArgs := []string{
+		"-c", "px-cluster",
+		"-x", "kubernetes",
+		"-b",
+		"-cloud_provider", "gce",
+		"-max_storage_nodes_per_zone", "3",
+		"-secret_type", "k8s",
+	}
+	actual, _ := driver.GetStoragePodSpec(cluster, fakeK8sNodes.Items[0].Name)
+	assert.ElementsMatch(t, expectedArgs, actual.Containers[0].Args)
+
+	// Reset preflight for other tests
+	coreops.SetInstance(coreops.New(fakek8sclient.NewSimpleClientset()))
+	err = preflight.InitPreflightChecker(k8sClient)
+	require.NoError(t, err)
+}
+
+func TestPodSpecWithCloudStorageSpecOnAzure(t *testing.T) {
+	fakeK8sNodes := &v1.NodeList{Items: []v1.Node{
+		{ObjectMeta: metav1.ObjectMeta{Name: "node1"}, Spec: v1.NodeSpec{ProviderID: "azure://node-id-1"}},
```

Comment:
Have we verified this in an azure cluster that it sets the providerID to `azure://...`?

Reply:
I didn't have an azure setup but this this test code is based off existing azure test code for "pkg/preflight/preflight_test.go".  Which I am assuming works.    Based on that the provider should get set.